### PR TITLE
Frontend auth

### PR DESCRIPTION
**Pull request by Besp1N**

### CODE REVIEW

File: sss-frontend/src/api/devicesApi.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import {getToken} from "../authUtils.ts";
+import {fetchSafe, HttpError} from "../http/fetch.ts";
+
+const API_URL = "http://localhost:8080";
+
+export async function fetchDevices(): Promise<{ id: string }[] | undefined> {
+    try {
```
Comment (PendolinoVoyager) [inline sss-frontend/src/api/devicesApi.ts:7]: unnecessary try/catch. Nothing throws here, fetchSafe return error

File: sss-frontend/src/api/devicesApi.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        const token = getToken();
+        if (!token) {
+            console.error("No token found. User might not be authenticated.");
+            return undefined;
+        }
+
+        const response = await fetchSafe<{ id: string }[]>(`${API_URL}/api/v1/device/`, {
```
Comment (PendolinoVoyager) [inline sss-frontend/src/api/devicesApi.ts]: see authUtils and addCredentials to attach token

File: sss-frontend/src/api/devicesApi.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                "Authorization": `Bearer ${token}`,
+            },
+        });
+
+        console.log(response);
```
Comment (PendolinoVoyager) [inline sss-frontend/src/api/devicesApi.ts]: remove this log

File: sss-frontend/src/components/activation_acc/MainContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { Box, Container, Typography } from "@mui/material";
+import InformativePanel from "./InformativePanel";
+import ActivationForm from "./ActivationForm";
+
+const MainContainer = () => (
```
Comment (PendolinoVoyager) [inline sss-frontend/src/components/activation_acc/MainContainer.tsx]: bad naming too generic

File: sss-frontend/src/components/pages/activationAccount.tsx
Comment (PendolinoVoyager) [inline sss-frontend/src/components/pages/activationAccount.tsx:1]: change the file and component name to ActivateAccountPage

File: sss-frontend/src/components/pages/deviceDetails.tsx
Comment (PendolinoVoyager) [inline sss-frontend/src/components/pages/deviceDetails.tsx:1]: same, change name to include page

File: sss-frontend/src/components/pages/registerPage.tsx
Comment (PendolinoVoyager) [inline sss-frontend/src/components/pages/registerPage.tsx:1]: If the only thing you show on a page is single component, make this component the page

File: sss-frontend/src/components/pages/resetPassword.tsx
Comment (PendolinoVoyager) [inline sss-frontend/src/components/pages/resetPassword.tsx:1]: change name to include page

**Review by PendolinoVoyager — CHANGES_REQUESTED** [review]
:(

File: sss-frontend/src/api/devicesApi.ts (hunk 1 above)
Comment (PendolinoVoyager) [reply sss-frontend/src/api/devicesApi.ts:7]: nvm

File: sss-frontend/src/api/devicesApi.ts (hunk 2 above)
Comment (Besp1N) [reply sss-frontend/src/api/devicesApi.ts]: done

File: sss-frontend/src/api/devicesApi.ts (hunk 1 above)
Comment (Besp1N) [reply sss-frontend/src/api/devicesApi.ts:7]: ok

File: sss-frontend/src/api/devicesApi.ts (hunk 3 above)
Comment (Besp1N) [reply sss-frontend/src/api/devicesApi.ts]: done

File: sss-frontend/src/components/activation_acc/MainContainer.tsx (hunk 4 above)
Comment (Besp1N) [reply sss-frontend/src/components/activation_acc/MainContainer.tsx]: Refactored

File: sss-frontend/src/components/pages/activationAccount.tsx
Comment (Besp1N) [reply sss-frontend/src/components/pages/activationAccount.tsx:1]: done

File: sss-frontend/src/components/pages/deviceDetails.tsx
Comment (Besp1N) [reply sss-frontend/src/components/pages/deviceDetails.tsx:1]: yes sir!

File: sss-frontend/src/components/pages/registerPage.tsx
Comment (Besp1N) [reply sss-frontend/src/components/pages/registerPage.tsx:1]: done

File: sss-frontend/src/components/pages/resetPassword.tsx
Comment (Besp1N) [reply sss-frontend/src/components/pages/resetPassword.tsx:1]: done!

**Review by Besp1N — COMMENTED** [review]
Done!

**Comment by PendolinoVoyager** [comment]
LGTM

**Pull request merged**